csm/common/filter.py

```diff
@@ -0,0 +1,100 @@
+# CORTX-CSM: CORTX Management web and CLI interface.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from cortx.utils.data.access.filters import Compare, And, Or
+from schematics.exceptions import ConversionError
+from csm.core.blogic.models import CsmModel
+import urllib.parse
+
+class Filter():
+    """Query to filter_object builder"""
+    @staticmethod
+    def parse_query(fields_of_filter):
+        """
+        Segregate list of key_val pairs and operands.
+        :param fields_of_filter: list of key_val pairs and operands like ['field1=val1', 'operand1', 'field2=val2']
+        :returns: dict(key_val_fields), dictionary of field(key) and value pairs
+        :returns: operations, Ordered list of operations to be performed on give field and value pairs
+        """
+        operations = []
+        key_val_fields = []
+        for key in fields_of_filter:
+            if key:
+                if key == "OR" or key == "AND":
+                    operations.append(key)
+                else:
+                    key_val_fields.append(map(str.strip, key.split('=', 1)))
+        return dict(key_val_fields), operations
+
+    @staticmethod
+    def validate_query_fields(query_fields, model):
```

Comment:
validate_query_fields --> _validate_query_fields. Remove static.

query_fields --> fields

Reply:
Addressed it by renaming variables.